src/sentry/snuba/entity_subscription.py

```diff
@@ -198,20 +204,36 @@ class MetricsCountersEntitySubscription(BaseEntitySubscription):
     dataset = Dataset.Metrics.value
     entity_key = EntityKey.MetricsCounters.value
 
-    def __init__(self, aggregate: str, extra_fields: Optional[Mapping[str, Any]] = None):
-        super().__init__(aggregate, extra_fields)
+    def __init__(
+        self, aggregate: str, time_window: int, extra_fields: Optional[Mapping[str, Any]] = None
+    ):
+        super().__init__(aggregate, time_window, extra_fields)
         self.aggregate = aggregate
         if not extra_fields or "org_id" not in extra_fields:
             raise InvalidQuerySubscription(
                 "org_id is a required param when "
                 "building snuba filter for a metrics subscription"
             )
         self.org_id = extra_fields["org_id"]
+        self.time_window = time_window
 
     def get_query_groupby(self) -> List[str]:
         session_status = tag_key(self.org_id, "session.status")
         return ["project_id", session_status]
 
+    def get_granularity(self):
+        # Both time_window and granularity are in seconds
+        # Time windows <= 1h -> Granularity 10s
+        # Time windows > 1h & <= 4h -> Granularity 60s
+        # Time windows > 4h -> Granularity 1 hour
```

Comment:
Wondering if there should be another granularity for > 4h <= 24hr?

Reply:
I will add it, probably makes sense 
Didn't add it initially because it was sort of assumed that the `else` corresponds to >4h and <=24h since we are validating that the maximum time window is 24h in the serializer, but it makes sense to be explicit about it